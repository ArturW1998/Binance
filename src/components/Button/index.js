import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

const Button = ({ onClick, value, id, className }) => {
  const classes = cn('btn', className);

  return (
    <div className="d-flex justify-content-center pt-2">
      <button className={classes} onClick={onClick} data-testid={id}>
        {value ? 'Disconnect' : 'Connect'}
      </button>
    </div>
  );
};

Button.propTypes = {
  onClick: PropTypes.func,
  value: PropTypes.bool,
  id: PropTypes.string,
  className: PropTypes.string,
};

Button.defaultProps = {
  onClick: () => {},
  value: true,
  id: '',
  className: '',
};

export default Button;
