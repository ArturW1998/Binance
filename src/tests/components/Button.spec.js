import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../components/Button';

describe('Button component', () => {
  describe('Button component value prop', () => {
    describe('Button component with positive value', () => {
      const props = {
        value: true,
      };

      const button = shallow(<Button {...props} />);

      it('renders properly', () => {
        expect(button).toMatchSnapshot();
      });

      it('renders Disconnect', () => {
        expect(button.find('button').text()).toEqual('Disconnect');
      });
    });

    describe('Button component with negative value', () => {
      const props = {
        value: false,
      };

      const button = shallow(<Button {...props} />);

      it('renders properly', () => {
        expect(button).toMatchSnapshot();
      });

      it('renders Connect', () => {
        expect(button.find('button').text()).toEqual('Connect');
      });
    });
  });

  describe('Button component onClick prop', () => {
    const mockOnClick = jest.fn();

    const props = {
      onClick: mockOnClick,
    };

    const button = shallow(<Button {...props} />);

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    describe('when clicking the button', () => {
      beforeEach(() => {
        button.find('button').simulate('click');
      });

      it('dispatches the `onClick()` method it receives from props', () => {
        expect(mockOnClick).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('Button component with id', () => {
    const props = {
      id: 'Connect',
    };

    const button = shallow(<Button {...props} />);

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('renders id', () => {
      expect(button.find('button').prop('data-testid')).toEqual(props.id);
    });
  });

  describe('Button component with className', () => {
    const props = {
      className: 'btn-warning',
    };

    const button = shallow(<Button {...props} />);

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('renders className', () => {
      expect(button.find('button').prop('className')).toEqual(
        `btn ${props.className}`,
      );
    });
  });

  describe('Button component with default props', () => {
    const mockOnClick = jest.fn();
    const props = {};

    const button = shallow(<Button {...props} />);

    it('renders properly', () => {
      expect(button).toMatchSnapshot();
    });

    it('renders default button value', () => {
      expect(button.find('button').text()).toEqual('Disconnect');
    });

    it('renders default button id', () => {
      expect(button.find('button').prop('data-testid')).toEqual('');
    });

    it('renders default button className', () => {
      expect(button.find('button').prop('className')).toEqual('btn');
    });

    describe('Button component onClick prop', () => {
      beforeEach(() => {
        button.find('button').simulate('click');
      });

      it('dispatches the `onClick()` method it receives from props', () => {
        expect(mockOnClick).toHaveBeenCalledTimes(0);
      });
    });
  });
});
