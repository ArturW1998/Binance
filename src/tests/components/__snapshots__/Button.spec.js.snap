// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button component onClick prop renders properly 1`] = `ShallowWrapper {}`;

exports[`Button component Button component value prop Button component with negative value renders properly 1`] = `ShallowWrapper {}`;

exports[`Button component Button component value prop Button component with positive value renders properly 1`] = `ShallowWrapper {}`;

exports[`Button component Button component with className renders properly 1`] = `ShallowWrapper {}`;

exports[`Button component Button component with default props renders properly 1`] = `ShallowWrapper {}`;

exports[`Button component Button component with id renders properly 1`] = `ShallowWrapper {}`;
